imap_processing/utils.py

```diff
@@ -342,34 +344,34 @@ def packet_file_to_datasets(
     variable_mapping: dict[int, set] = dict()
 
     # Set up the parser from the input packet definition
-    packet_definition = xtcedef.XtcePacketDefinition(xtce_packet_definition)
-    packet_parser = parser.PacketParser(packet_definition)
+    packet_definition = definitions.XtcePacketDefinition(xtce_packet_definition)
 
     with open(packet_file, "rb") as binary_data:
-        packet_generator = packet_parser.generator(binary_data)
+        packet_generator = packet_definition.packet_generator(binary_data)
         for packet in packet_generator:
             apid = packet.header["PKT_APID"].raw_value
             if apid not in data_dict:
                 # This is the first packet for this APID
                 data_dict[apid] = collections.defaultdict(list)
                 datatype_mapping[apid] = dict()
-                variable_mapping[apid] = packet.data.keys()
-            if variable_mapping[apid] != packet.data.keys():
+                variable_mapping[apid] = packet.user_data.keys()
+            if variable_mapping[apid] != packet.user_data.keys():
                 raise ValueError(
                     f"Packet fields do not match for APID {apid}. This could be "
                     f"due to a conditional packet definition in the XTCE, while this "
                     f"function currently only supports flat packet definitions."
-                    f"\nExpected: {variable_mapping[apid]},\ngot: {packet.data.keys()}"
+                    f"\nExpected: {variable_mapping[apid]},\n"
+                    f"got: {packet.user_data.keys()}"
                 )
 
             # TODO: Do we want to give an option to remove the header content?
-            packet_content = packet.data | packet.header
+            packet_content = packet.user_data | packet.header
 
             for key, value in packet_content.items():
                 val = value.raw_value
```

Comment:
packet_content is just the packet here now, no need to combine them together anymore.
```suggestion
            for key, value in packet.items():
                val = value if use_derived_value else value.raw_value
```

Reply:
I can't make all of these change because we had reason for flipping these two out. We use first key as `time_key` to calculate epoch time. I can make the if else statement change though.